datapackage/resource.py

```diff
@@ -279,6 +279,7 @@ def iter(self):
                 parser_class = tabulator.parsers.CSV
             try:
                 table = tabulator.topen(data_path_or_url, with_headers=True,
+                                        encoding=self.metadata.get('encoding'),
```

Comment:
tabulator actually runs an encoding detection routine, so passing nothing actually means 'try to detect the encoding on your own'. Passing something means 'just use this encoding'. From what I know, UTF8 is not assumed...

As for a test, I guess we could try to pass a wrong encoding and make sure it fails. wdyt?